### PR TITLE
Issue #146 - ArgoCD applications should have a rolled up health status

### PR DESCRIPTION
**Pull request by alexmt**
PR adds health information to Application CRD and root kubernetes resources ( e.g. deployment, service ). Health status for child resources is was not added since root resources already has rolled up health status, deployment is not healthy if pod crashing etc.

### CODE REVIEW

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ type ApplicationStatus struct {
 	ComparisonResult  ComparisonResult     `json:"comparisonResult" protobuf:"bytes,1,opt,name=comparisonResult"`
 	RecentDeployments []DeploymentInfo     `json:"recentDeployments" protobuf:"bytes,2,opt,name=recentDeployment"`
 	Parameters        []ComponentParameter `json:"parameters,omitempty" protobuf:"bytes,3,opt,name=parameters"`
+	HealthState       HealthState          `json:"healthState,omitempty" protobuf:"bytes,4,opt,name=healthState"`
```
Comment (jessesuen) [inline pkg/apis/application/v1alpha1/types.go]: Should this be HealthStatus instead of state? I tend to follow [this rule](https://stackoverflow.com/a/11229919):

status == how are you? [good/bad]
state == what are you doing? [resting/working]

Also, for the field name, lets shorten this to just "health". The type name can still be HealthStatus.

Then when accessing it will be app.Status.Health.Status.

**Review by jessesuen — COMMENTED** [review]
Implementation looks fine. Terminology I think "status" makes more sense than "state", and to shorten field name to "health".

File: pkg/apis/application/v1alpha1/types.go (hunk 1 above)
Comment (alexmt) [reply pkg/apis/application/v1alpha1/types.go]: Good suggestion! done.

**Review by merenbach — APPROVED** [review]
